src/index.js

```diff
@@ -0,0 +1,34 @@
+const { formatDate, isDateInList, bankHolidayList, validCountry } = require('./utils');
+const request = require('superagent');
+
+const isDateABankHoliday = async function(date, countries) {
+
+    if (!date && !countries) {
+        throw new Error('No arguments passed');
+    }
+
+    if (!countries) {
+        throw new Error('Countries not passed');
+    }
+
+    if (!Array.isArray(countries)) {
+        throw new Error('Countries must be an array');
+    }
+
+    if (!validCountry(countries)) {
+        throw new Error('Country not on the list');
+    }
+
+    date = formatDate(date);
```

Comment:
I think you need to perform a check to ensure the date string is in the correct format `DD-MM-YYYY`. What happens if I pass an empty string or accidentally miss off a `Y`?

Reply:
@paulgain point. I can do `moment().isValid()` to check

Reply:
or a regex

Reply:
Whatever you decide ensure the user has a good error message.